delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/directives/DeltaDirectives.scala

```diff
@@ -20,29 +19,40 @@ import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.JsonLd
 import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
 import ch.epfl.bluebrain.nexus.delta.rdf.utils.JsonKeyOrdering
 import ch.epfl.bluebrain.nexus.delta.rdf.RdfMediaTypes._
+import ch.epfl.bluebrain.nexus.delta.sdk.AkkaSource
 import ch.epfl.bluebrain.nexus.delta.sdk.marshalling.{HttpResponseFields, JsonLdFormat, RdfMarshalling}
 import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults
 import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults._
 import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Event}
 import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.HeadersUtils
 import fs2.Stream
 import monix.bio.{IO, Task, UIO}
 import monix.execution.Scheduler
 import streamz.converter._
 
+import java.nio.charset.StandardCharsets
 import scala.util.Try
 
 object DeltaDirectives extends UriDirectives with RdfMarshalling {
 
   // order is important
-  private val mediaTypes =
-    Seq(
+  val mediaTypes: List[MediaType.WithFixedCharset] =
+    List(
       `application/ld+json`,
       `application/json`,
       `application/n-triples`,
       `text/vnd.graphviz`
     )
 
+  /**
+    * Filename, file contentType and file data
+    */
+  type FileData = (String, ContentType, AkkaSource)
```

Comment:
A case class in the model package would be clearer

Reply:
I didn't want to bother wrapping things up, since it isn't broadly used, but I can do that